host/async_wf_test.go

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+//go:build !race && asyncwfintegration
+// +build !race,asyncwfintegration
+
+/*
+To run locally:
+
+1. Stop the previous run if any
+
+	docker-compose -f docker/buildkite/docker-compose-local-async-wf.yml down
+
+2. Build the integration-test-async-wf image
+
+	docker-compose -f docker/buildkite/docker-compose-local-async-wf.yml build integration-test-async-wf
+
+3. Run the test in the docker container
+
+	docker-compose -f docker/buildkite/docker-compose-local-async-wf.yml run --rm integration-test-async-wf
+
+4. Full test run logs can be found at test.log file
+*/
+package host
+
+import (
+	"flag"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/pborman/uuid"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/clock"
+	"github.com/uber/cadence/common/dynamicconfig"
+	"github.com/uber/cadence/common/persistence"
+	pt "github.com/uber/cadence/common/persistence/persistence-tests"
+	"github.com/uber/cadence/common/types"
+
+	_ "github.com/uber/cadence/common/asyncworkflow/queue/kafka" // needed to load kafka asyncworkflow queue
+)
+
+type AsyncWFIntegrationSuite struct {
```

Comment:
Can you move the definition to `test_suites.go` file?
We need to reuse the definition in monorepo

Reply:
why do we need to define this struct in `test_suites.go`? I was following pinot_test.go mainly

Reply:
synced offline.